processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -109,6 +118,22 @@ public SearchQueryRunner(Segment segment)
       final ImmutableBitmap baseFilter =
           filter == null ? null : filter.getBitmapIndex(new ColumnSelectorBitmapIndexSelector(bitmapFactory, index));
 
+      MutableBitmap timeBitmap = bitmapFactory.makeEmptyMutableBitmap();
+      final Column timeColumn = index.getColumn(Column.TIME_COLUMN_NAME);
+      final GenericColumn timeValues = timeColumn.getGenericColumn();
+
+      for (int i = 0; i < timeValues.length(); i++)
+      {
+        long time = timeValues.getLongSingleValueRow(i);
+        if (interval.contains(time))
+        {
+          timeBitmap.add(i);
+        }
+      }
+      final ImmutableBitmap finalTimeBitmap = bitmapFactory.makeImmutableBitmap(timeBitmap);
+      final ImmutableBitmap timeFilteredBitmap =
+          (baseFilter == null) ? finalTimeBitmap : finalTimeBitmap.intersection(baseFilter);
+
```

Comment:
Most query runners would be ran on segment which is subset of interval in the query, which will make `timeBitmap` all true. We can optimize this part further.